### ISSUE TITLE
Diff between Kustomize and Helm

### ISSUE THREAD

**Issue by lindhe** [metallb/metallb#2042]
### MetalLB Version

0.13.10

### Deployment method

Not relevant

### Main CNI

n/a

### Describe the bug

There's a huge diff between the manifest produced by the Helm chart and that produced by Kustomize. Is that a bug? I was going to migrate from Helm to Kustomize for MetalLB, and was taken aback by the huge `kubectl diff` I saw. Large parts are unchanged, but large parts are changed (mainly additions).

This makes me think that I've either used the wrong variant in Kustomize, that I'm missing some crucial values for Helm, or that there is a feature gap between the Helm and Kustomize installations. 

If there's no bug here, just me not understanding the installation alternatives, I think that [the installation instructions](https://metallb.org/installation/) should be clarified.

### To Reproduce

1. First, create a `kustomization.yaml`:

    ```shell
    cat <<EOF > kustomization.yaml
    apiVersion: kustomize.config.k8s.io/v1beta1
    kind: Kustomization
    namespace: metallb-system
    resources:
    - github.com/metallb/metallb/config/native?ref=v0.13.10
    EOF
    ```

2. Now, compare the outputs. This is a bit tricky, since YAML files are not necessarily ordered in the same way. To compare the manifests semantically, I did the following:

    ```shell
    helm install -n metallb-system metallb metallb/metallb --version 0.13.10
    kustomize build . | kubectl diff -f -
    ```

    But to just get the files so one can compare with other tools (e.g. by manual inspection), this should suffice:

    ```shell
    kustomize build . > from-kustomize.yaml
    helm template -n metallb-system metallb metallb/metallb --version 0.13.10 > from-helm.yaml
    ```

### Expected Behavior

Given the right parameters, I would expect the Helm chart to produce the exact same output as the Kustomization does (except for very minor things like setting the label `app.kubernetes.io/managed-by=Helm`, some resource name here or there, etc.).

### Additional Context

I am not sure that I use the right values to make the Helm chart produce the same content as the Kustomization does. Nor am I sure I use the correct Kustomization, there are apparently at least two (native and frr).

I've tried both `github.com/metallb/metallb/config/native?ref=v0.13.10` and `github.com/metallb/metallb/config/frr?ref=v0.13.10` using Kustomize, and it's similar results compared to the Helm chart.

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### I've read and agree with the following

- [X] I've checked all open and closed issues and my issue is not there.
- [X] This bug is reproducible when deploying MetalLB from the main branch
- [X] I have read the [troubleshooting guide](https://metallb.universe.tf/troubleshooting/) and I am still not able to make it work
- [X] I checked the logs and MetalLB is not discarding the configuration as not valid
- [X] I enabled the debug logs, collected the information required from the cluster using the [collect script](https://raw.githubusercontent.com/metallb/metallb/main/troubleshooting/collect.sh) and will attach them to the issue
- [X] I will provide the definition of my service and the related endpoint slices and attach them to this issue

**Comment by fedepaol** [comment]
Thanks for raising this and yes, unfortunately we don't currently have a migration path between the two alternatives, which grew in an non-organic manner, and I don't foresee any plan to align the two.

Both deployment ways are supported and afaik work, and are tested in CI.
I am going to close this issue as not planned.

**Comment by lindhe** [comment]
Hi! Thanks for the reply!

Choosing to have a discrepancy between the two is understandable given the situation. But having match version numbers for them is, in that case, very misleading. Perhaps that can be remedied?

**Comment by lindhe** [comment]
Or at least there could be a comment about it in the documentation, explaining which one is more up-to-date. Which one is most up-to-date?

**Comment by fedepaol** [comment]
> Hi! Thanks for the reply!
> 
> Choosing to have a discrepancy between the two is understandable given the situation. But having matching version numbers for them is, in that case, very misleading. Perhaps that can be remedied?

Not sure I understood what you mean. Both refer to the same metallb version and deploy the same binaries

**Comment by lindhe** [comment]
You are right to not understand what I wrote, because I got it all wrong. What I originally meant was that the Helm chart's `version` matches the release version of the rest of the manifests, which is confusing.

But I realize that it's not so clear cut. The `appVersion` do match the release version because, as you say, they deploy the same binaries.

Having matching `version` and `appVersion` is a bit odd, but there's no matching version numbering for the manifests/kustomize so it makes no sense to compare the two. 

Yeah, it's confusing all around this one. 😄 I guess that's what one gets for running pre-release software.
